### PR TITLE
fix(nms): fix CBSD pagination

### PR DESCRIPTION
**Pull request by ivan-freedomfi**
## Summary

Fix CBSD pagination by supplying a proper data function for the Material
Table as per docs:
https://material-table.com/#/docs/features/remote-data

(CBSD pagination wasn't working - it was always showing the 1st page).

Also, remove the `<LoadingFiller>`, because it makes the
`material-table` re-mount on each cbsd page fetch, causing it to request
the first page (it loses state), no matter which page you clicked.

Signed-off-by: Ivan Sergiienko <ivan@freedomfi.com>

## Test Plan

1. Open cbsds list page
2. Click pagination: next/previous, first/last
3. It all works: pages are changed correctly, new data is shown (previously it was stuck on the 1st page always).

cbsd-pagination

### CODE REVIEW

File: nms/app/context/CbsdContext.tsx [hunk 1]
```diff
@@ -160,6 +161,11 @@ export async function remove(params: {networkId: string; cbsdId: number}) {
   });
 }
 
+interface PaginationOptions {
```
Comment (Neudrino) [inline nms/app/context/CbsdContext.tsx]: We do not use Interfaces anywhere in the code. If it is a type definition, use of `type` is preferred.

File: nms/app/context/CbsdContext.tsx [hunk 2]
```diff
@@ -174,22 +180,20 @@ export function CbsdContextProvider({
     cbsds: [],
     total_count: 0,
   });
-  const [paginationOptions, setPaginationOptions] = useState<{
-    page: number;
-    pageSize: number;
-  }>({
-    page: 0,
+  const [paginationOptions, setPaginationOptions] = useState<PaginationOptions>({
+    page: -1,
```
Comment (Neudrino) [inline nms/app/context/CbsdContext.tsx]: Is `-1` instead of `0` valid input?

File: nms/app/views/domain-proxy/CbsdsTable.tsx [hunk 3]
```diff
@@ -125,32 +125,46 @@ function CbsdsTable(props: WithAlert) {
...
+  }, [ctx.state.page, ctx.state.pageSize, ctx.state.totalCount]);
 
   const currentCbsd = useMemo(() => {
     return ctx.state.cbsds
       ? ctx.state.cbsds.find(({id}) => id === currentRow.id)
       : undefined;
   }, [ctx.state.cbsds, currentRow]);
 
-  if (ctx.state.isLoading) return <LoadingFiller />;
-
```
Comment (Neudrino) [inline nms/app/views/domain-proxy/CbsdsTable.tsx:153]: Why don't we need the loading filler any more?

Comment (ivan-freedomfi) [reply nms/app/views/domain-proxy/CbsdsTable.tsx:153]: [material-table](https://material-table.com/#/docs/features/remote-data) has it's own spinner, it can be seen in the docs example, and you can also see a gif in the PR description, the loader is there.

So I believe this extra spinner is not needed here.
Moreover, with this spinner the Table will be removed from the DOM on each cbsd page load. This causes the table to mount again, and make a request for the first page every time (this was part of the bug).

So basically the library was used in a wrong way, and this loader is not needed, I believe.

File: nms/app/context/CbsdContext.tsx (hunk 1 above)
Comment (ivan-freedomfi) [reply nms/app/context/CbsdContext.tsx]: Changed to `type`.

File: nms/app/views/domain-proxy/CbsdsTable.tsx (hunk 3 above)
Comment (ivan-freedomfi) [reply nms/app/views/domain-proxy/CbsdsTable.tsx:153]: I've now added this explanation to the commit message.

File: nms/app/context/CbsdContext.tsx (hunk 2 above)
Comment (ivan-freedomfi) [reply nms/app/context/CbsdContext.tsx]: No :) it was a hack. I've figured out how to make it properly now :)
So now initial page is `0`.

**Review by Neudrino — APPROVED** [review]
Thanks a lot for this contribution and fix! :100:

**Comment by mpfirrmann** [comment]
@ivan-freedomfi, is there a reason this was never merged?

**Comment by voisey** [comment]
Hey @ivan-freedomfi, thanks for the PR! Could you please rebase this and then set the "ready2merge" label if you think it's good to go?

**Review by jordanvrtanoski — APPROVED** [review]

**Comment by shahrahman-fb** [comment]
Thank you for this contribution\! We're closing this PR as part of a repository cleanup effort — it has had no activity for an extended period and may have conflicts with the current codebase.

If this work is still relevant, you're welcome to rebase and resubmit against current `master`. We're actively reviewing and merging contributions again.

See: #15833 (Phase 1 cleanup), #15828 (tracking issue)
